### PR TITLE
Engine api optimizations

### PR DESCRIPTION
**Pull request by lucassaldanha**
(same as https://github.com/Consensys/teku/pull/10457)

## PR Description

This PR optimises the deserialization of some Engine API requests. Most requests to return blobs return large hex strings as the data's byte representation, so they require some extra care on deserialization to avoid unnecessary overhead.

Summary of changes:
- Updated the deserialization logic on Bytes-related Jackson deserializers used on Engine API calls. Stopped using `getValueAsString()` to avoid allocating intermediate strings and added a more streamlined lookup table for hex to byte conversion.
- Avoiding excessive Java reflection calls with Jackson
    - By explicitly using `@JsonCreator`, and
    - By using Jackson's `Blackbird` module.

I added both JMH benchmarks used to enable easier testing for further improvements.

## Fixed Issue(s)
N/A

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

---

> [!NOTE]
> **Medium Risk**
> Touches Engine API JSON deserialization (hex bytes parsing and Web3j `ObjectMapper` configuration), so regressions could break execution client communication despite added tests/benchmarks.
> 
> **Overview**
> **Improves Engine API deserialization performance** by rewriting `Bytes*` Jackson deserializers to share a new `AbstractBytesDeserializer` that parses hex directly from the parser’s char buffer (avoiding `getValueAsString()` allocations) and uses a lookup-table based hex decoder.
> 
> **Reduces Jackson reflection overhead** by adding `@JsonCreator` to key Engine API schema constructors and registering Jackson `BlackbirdModule` in `Web3jHttpClient` (plus adding the `jackson-module-blackbird` dependency).
> 
> Adds JMH benchmarks for bytes deserializers and full Engine API payload/blob response deserialization, adds unit tests covering hex parsing edge-cases, and updates an SSZ benchmark to consume results via `Blackhole` to avoid dead-code elimination.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit e7e146be0899c10297aaf0a929ae2c9f5e90bec8. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

**Review by rolfyone — APPROVED** [review]
LGTM

File: ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/serialization/AbstractBytesDeserializer.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    }
+    final byte[] result = new byte[hexLen / 2];
+    int pos = start;
+    for (int i = 0; i < result.length; i++, pos += 2) {
+      final int hi = hexValue(chars[pos]);
+      final int lo = hexValue(chars[pos + 1]);
+      if ((hi | lo) < 0) {
+        throw new IllegalArgumentException(
+            "Invalid hex character at position " + pos + ": '" + chars[pos] + "'");
+      }
```
[the remark this answers is not in the document]
Comment (gfukushima) [reply ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/serialization/AbstractBytesDeserializer.java:86]: worth looking into this case before merging?

[the remark this answers is not in the document]
Comment (tbenr) [reply ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/serialization/AbstractBytesDeserializer.java:86]: this is a valid issue

**Comment by lucassaldanha** [comment]
Block and Attestation inclusions didn't seem to have any hit with the updated client:

image

image

Using getBlobsV2 as a reference for request time, the observed improvement isn't huge (but also not negligible):

image

File: ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/serialization/Bytes32Deserializer.java [hunk 2]
```diff
@@ -13,17 +13,12 @@
...
 
-public class Bytes32Deserializer extends JsonDeserializer<Bytes32> {
+public class Bytes32Deserializer extends AbstractBytesDeserializer<Bytes32> {
 
   @Override
-  public Bytes32 deserialize(final JsonParser p, final DeserializationContext ctxt)
-      throws IOException {
-    return Bytes32.fromHexStringStrict(p.getValueAsString());
+  protected Bytes32 fromRawBytes(final byte[] bytes) {
+    return Bytes32.wrap(bytes);
```
[the remark this answers is not in the document]
Comment (tbenr) [reply ethereum/executionclient/src/main/java/tech/pegasys/teku/ethereum/executionclient/serialization/Bytes32Deserializer.java:22]: the `wrap` method checks the bytes length to be exactly the expected size.
The `0x` isn't a big deal.
